### PR TITLE
Attempts to fix the holsters no one uses (again)

### LINKED ISSUE #9096: MR25 harness has an error sprite when loaded
### Description

The PO only MR25 harness has the ERROR sprite if you actually put a gun in it.
Otherwise works normally.

### Test Merges

Serb down, but probably doesn't matter.

### Reproduction Steps

Hassle Req for an MR25/get it as seasonal
Try to be a tacticool dual MR25 PO
Cry at the error sprite

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Notamaniac**
## About The Pull Request

Stops the shotgun holsters and SMG holsters from showing an error sprite or just plain vanishing.

Closes #9096

## Why It's Good For The Game

For that rare specimen of a PO who uses their MR25 holster.

## Changelog
:cl:
fix: The sprites for the MR25 holster should work now. What, you didn't know we still had those?
/:cl:

### CODE REVIEW

File: code/game/objects/items/storage/large_holster.dm [hunk 1]
```diff
@@ -13,14 +13,17 @@
 	var/base_icon = "m37_holster"
 	var/drawSound = 'sound/weapons/guns/misc/rifle_draw.ogg'
 
-
 /obj/item/storage/large_holster/update_icon()
+	..()
```
Comment (hyper2snyper) [inline code/game/objects/items/storage/large_holster.dm:17]:
```suggestion
	. = ..()
```

File: code/game/objects/items/storage/large_holster.dm [hunk 2]
```diff
@@ -13,14 +13,17 @@
...
 	var/mob/user = loc
-	icon_state = "[base_icon][contents.len?"_full":""]"
-	item_state = icon_state
 	if(istype(user)) user.update_inv_back()
```
Comment (hyper2snyper) [inline code/game/objects/items/storage/large_holster.dm:19]: put these all under the same if().
dont need three identical ifs

File: code/game/objects/items/storage/large_holster.dm [hunk 3]
```diff
@@ -53,10 +56,9 @@
 	)
 
 /obj/item/storage/large_holster/m37/full/Initialize()
-	. = ..()
-	icon_state = "m37_holster_full"
 	new /obj/item/weapon/gun/shotgun/pump(src)
-	base_icon = icon_state
+	update_icon()
+	. = ..()
```
Comment (hyper2snyper) [inline code/game/objects/items/storage/large_holster.dm:61]:
```suggestion
	return ..()	
```

**Review by hyper2snyper — CHANGES_REQUESTED** [review]

**Comment by Notamaniac** [comment]
@hyper2snyper 
Done

**Review by BraveMole — APPROVED** [review]

**Pull request merged**